Tests/NumLib/TestMeshComponentMap.cpp

```diff
@@ -215,3 +215,82 @@ TEST_F(NumLibMeshComponentMapTest, DISABLED_SubsetOfNodesByLocation)
             cmap_subset.getGlobalIndex(l, comp1_id));
     }
 }
+
+#ifndef USE_PETSC
+TEST_F(NumLibMeshComponentMapTest, MulticomponentVariable)
+#else
+TEST_F(NumLibMeshComponentMapTest, DISABLED_MulticomponentVariable)
+#endif
+{
+    cmap = new MeshComponentMap(components,
+        NumLib::ComponentOrder::BY_LOCATION);
+
+    // Select some nodes from the full mesh.
+    std::array<std::size_t, 3> const ids = {{ 0, 5, 9 }};
+    std::vector<MeshLib::Node*> some_nodes;
+    for (std::size_t id : ids)
+        some_nodes.push_back(const_cast<MeshLib::Node*>(mesh->getNode(id)));
+
+    MeshLib::MeshSubset some_nodes_mesh_subset(*mesh, &some_nodes);
+
+    auto selected_component = std::unique_ptr<MeshLib::MeshSubsets>{
+        new MeshLib::MeshSubsets{&some_nodes_mesh_subset}};
```

Comment:
Minor: actually there is probably no need for a unique_ptr. A normal object would suffice.